src/Entities/Player.cpp

```diff
@@ -426,12 +428,75 @@ void cPlayer::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 
 	Super::Tick(a_Dt, a_Chunk);
 
+	float Width;
+	float Height;
+	if (m_IsElytraFlying)
+	{
+		Width = 0.6F;
+		Height = 0.6F;
+	}
+	else if (m_bIsInBed)
+	{
+		Width = 0.2F;
+		Height = 0.2F;
+	}
+	else if (m_IsCrouched)
+	{
+		Width = 0.6F;
+		Height = 1.65F;
+	}
+	else
+	{
+		Width = 0.6F;
+		Height = 1.8F;
+	}
+
+	if ((Width != GetWidth()) || (Height != GetHeight()))
+	{
+		SetSize(Width, Height);
+		float EyeHeight = 1.62F;
+
+		if (IsInBed())
+		{
+			EyeHeight = 0.2F;
+		}
+		else if (IsCrouched())
+		{
+			EyeHeight -= 0.08F;
+		}
+		else if (IsElytraFlying())
+		{
+			EyeHeight = 0.4F;
+		}
+
+		m_RelativeEyeHeight = EyeHeight;
+		SetStance(GetPosY()+EyeHeight);
+	}
+
 	// Handle charging the bow:
 	if (m_IsChargingBow)
 	{
 		m_BowCharge += 1;
 	}
 
+
+	/** Elytra flying */
+	if (m_IsElytraFlying && !IsOnGround() && !IsRiding())
+	{
+		if ((m_TicksElytraFlying + 1) % 20 == 0)
+		{
+			UseItem(cInventory::invArmorOffset + 1);
+		}
+
+		m_TicksElytraFlying++;
+	}
+	else if (m_IsElytraFlying)
+	{
+		m_TicksElytraFlying = 0;
+		m_IsElytraFlying = false;
+		m_World->BroadcastEntityMetadata(*this);
+	}
```

Comment:
Use IsElytraFlying() rather than m_IsElytraFlying to be consistent. Also is there a reason SetElytraFlight is not used here?